### PR TITLE
refactor(cf): use fetch API for file uploads

### LINKED ISSUE #593: Use fetch for file uploads
It's 2019 not 1997... :wink: 

https://github.com/projectcaluma/ember-caluma/blob/c424c0c06fd5707d73f618818ecbdfb74c2fc8db/addon/components/cf-field/input/file.js#L34-L54

### PR DESCRIPTION
**Pull request by luytena**
Replace XMLHttpRequest implementation for file
uploads with fetch API.

Resolves #593

### CODE REVIEW

File: addon/components/cf-field/input/file.js [hunk 1]
```diff
@@ -72,6 +67,7 @@ export default Component.extend({
       }
     },
 
+    @waitFor
```
Comment (anehx) [inline addon/components/cf-field/input/file.js:70]: Using a test helper in a component - why?

Comment (luytena) [reply addon/components/cf-field/input/file.js:70]: The `await triggerEvent(...)` even though it apparently uses `settled()` internally it doesn't actually wait for the asynchronous function to terminate. From what I've read it doesn't realize that it has to wait for that particular promise and that's where the emberjs `waitFor` test helper decorator gives it that information. Is there a solution that can be used within the test that ensures that it waits for the `save` function to terminate?

Comment (luytena) [reply addon/components/cf-field/input/file.js:70]: Using the `waitUntil` test helper in the test seems to work as well.

Comment (anehx) [reply addon/components/cf-field/input/file.js]: Seems more reasonable

**Review by anehx — APPROVED** [review]
Good work!

**Pull request merged**